dev/src/expression.ts

```diff
@@ -0,0 +1,1425 @@
+  constructor(
+    private expr: Expr,
+    private prefix: Expr
+  ) {
+    super('starts_with', [expr, prefix]);
+  }
+  filterable = true as const;
+}
+
+class EndsWith extends Function implements FilterCondition {
+  constructor(
+    private expr: Expr,
+    private suffix: Expr
+  ) {
+    super('ends_with', [expr, suffix]);
+  }
+  filterable = true as const;
+}
+
+class ToLowercase extends Function {
+  constructor(private expr: Expr) {
+    super('to_lowercase', [expr]);
+  }
+}
+
+class ToUppercase extends Function {
+  constructor(private expr: Expr) {
+    super('to_uppercase', [expr]);
+  }
+}
+
+class Trim extends Function {
+  constructor(private expr: Expr) {
+    super('trim', [expr]);
+  }
+}
+
+class StrConcat extends Function {
+  constructor(
+    private first: Expr,
+    private rest: Expr[]
+  ) {
+    super('str_concat', [first, ...rest]);
+  }
+}
+
+class MapGet extends Function {
+  constructor(map: Expr, name: string) {
+    super('map_get', [map, Constant.of(name)]);
+  }
+}
+
+class Count extends Function implements Accumulator {
+  accumulator = true as const;
+  constructor(
+    private value: Expr | undefined,
+    private distinct: boolean
+  ) {
+    super('count', value === undefined ? [] : [value]);
+  }
+}
+
+class Sum extends Function implements Accumulator {
+  accumulator = true as const;
+  constructor(
+    private value: Expr,
+    private distinct: boolean
+  ) {
+    super('sum', [value]);
+  }
+}
+
+class Avg extends Function implements Accumulator {
+  accumulator = true as const;
+  constructor(
+    private value: Expr,
+    private distinct: boolean
+  ) {
+    super('avg', [value]);
+  }
+}
+
+class Min extends Function implements Accumulator {
+  accumulator = true as const;
+  constructor(
+    private value: Expr,
+    private distinct: boolean
+  ) {
+    super('min', [value]);
+  }
+}
+
+class Max extends Function implements Accumulator {
+  accumulator = true as const;
+  constructor(
+    private value: Expr,
+    private distinct: boolean
+  ) {
+    super('max', [value]);
+  }
+}
+
+class CosineDistance extends Function {
+  constructor(
+    private vector1: Expr,
+    private vector2: Expr
+  ) {
+    super('cosine_distance', [vector1, vector2]);
+  }
+}
+
+class DotProductDistance extends Function {
+  constructor(
+    private vector1: Expr,
+    private vector2: Expr
+  ) {
+    super('dot_product', [vector1, vector2]);
+  }
+}
+
+class EuclideanDistance extends Function {
+  constructor(
+    private vector1: Expr,
+    private vector2: Expr
+  ) {
+    super('euclidean_distance', [vector1, vector2]);
+  }
+}
+
+export function add(left: Expr, right: Expr): Add;
+export function add(left: Expr, right: any): Add;
+export function add(left: string, right: Expr): Add;
+export function add(left: string, right: any): Add;
+export function add(left: Expr | string, right: Expr | any): Add {
+  const normalizedLeft = typeof left === 'string' ? Field.of(left) : left;
```

Comment:
Great candidate for a helper `Field.fromArgument(left)`. Similar to `FieldPath.fromArgument(stringOrFieldPath)`

Reply:
Will do.